airflow/providers/amazon/aws/sensors/athena.py

```diff
@@ -49,20 +50,20 @@ class AthenaSensor(BaseSensorOperator):
 
     @apply_defaults
     def __init__(self,
-                 query_execution_id,
-                 max_retires=None,
-                 aws_conn_id='aws_default',
-                 sleep_time=10,
-                 *args, **kwargs):
+                 query_execution_id: str,
+                 max_retires: Optional[int] = None,
+                 aws_conn_id: str = 'aws_default',
+                 sleep_time: int = 10,
+                 *args: Any, **kwargs: Any) -> None:
         super().__init__(*args, **kwargs)
         self.aws_conn_id = aws_conn_id
         self.query_execution_id = query_execution_id
         self.sleep_time = sleep_time
         self.max_retires = max_retires
-        self.hook = None
+        self.hook = self.get_hook()
```

Comment:
This should be None bye default to avoid initializing hooks in Operators init.

Otherwise, whenever the scheduler parses the DAG files, DB calls are made to the database.

You could do the following to fix mypy errors:

```suggestion
        self.hook: Optional[AWSAthenaHook] = None
```